overlord/devicestate/firstboot_test.go

```diff
@@ -445,9 +442,18 @@ func (s *FirstBootTestSuite) TestPopulateFromSeedHappy(c *C) {
 func (s *FirstBootTestSuite) TestPopulateFromSeedMissingBootloader(c *C) {
 	chg := s.makeBecomeOpertionalChange(c)
 	st := s.overlord.State()
+
+	// run the change, we cannot use s.overlord.Settle() here as each
+	// "DeviceManager.Ensure()" will call ensureSeedYaml which will
+	// create a new "seed" change so things never settle down
+	s.overlord.Loop()
+	defer s.overlord.Stop()
+	for !chg.IsReady() {
+		// nothing
+	}
+
```

Comment:
this is strange, why would the changes affect first boot stuff?

Reply:
I played a bit,  the "issue" seems to be that prerequisites not having an undo handler make it so that the change is ready earlier and so an ensure cycle triggers agains seed. I don't care too much either way, but I'm not a fan of the Overlord.Loop() in this approach. I think firstboot_test.go are the only tests that use a full overlord in *state/*_test.go.  I have a plan since a while to have a overlord.Mock() that would give us a bit more control when we need to setup more than one manager, maybe it's time for me to look into that. It hopefully would help kill all the various settle helpers we have around as well but need to try.